File: pact-consumer/src/main/java/com/example/SuppliesClient.java
package com.example;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.module.paramnames.ParameterNamesModule;
import org.apache.http.HttpResponse;
import org.apache.http.client.fluent.Request;
import org.apache.http.util.EntityUtils;

import java.io.IOException;
import java.time.LocalDate;
import java.util.List;

public class SuppliesClient {
    private final String baseUrl;
    private final ObjectMapper mapper;

    public SuppliesClient(String baseUrl) {
        this.baseUrl = baseUrl;

        mapper = createMapper();
    }

    public List<Supply> getFor(LocalDate day) {
        try {
            HttpResponse response = Request.Get(baseUrl + "/supplies?day=" + day).execute().returnResponse();
            if (response.getStatusLine().getStatusCode() >= 300) {
                throw errorFor(response);
            }
            return mapper.readValue(
                    EntityUtils.toString(response.getEntity()),
                    new TypeReference<List<Supply>>() { }
            );
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    private RuntimeException errorFor(HttpResponse response) throws IOException {
        return new RuntimeException("request failed with HTTP " + response.getStatusLine().getStatusCode() + ":\n"
                + EntityUtils.toString(response.getEntity()));
    }

    private static ObjectMapper createMapper() {
        ObjectMapper result = new ObjectMapper();
        result.registerModule(new ParameterNamesModule());
        return result;
    }
}

File: pact-consumer-kotlin/src/main/kotlin/com/example/SuppliesClient.kt
package com.example

import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.kotlin.KotlinModule
import org.apache.http.HttpResponse
import org.apache.http.client.fluent.Request
import org.apache.http.util.EntityUtils
import java.time.LocalDate

class SuppliesClient(private val baseUrl: String) {
    private val mapper = createMapper()

    fun getFor(day: LocalDate): List<Supply> {
        val response = Request.Get("$baseUrl/supplies?day=$day").execute().returnResponse()
        if (response.statusLine.statusCode >= 300) {
            throw errorFor(response)
        }
        return mapper.readValue(
                EntityUtils.toString(response.entity),
                object : TypeReference<List<Supply>>() {}
        )
    }

    private fun errorFor(response: HttpResponse): RuntimeException {
        return RuntimeException("request failed with HTTP " + response.statusLine.statusCode + ":\n" +
                EntityUtils.toString(response.entity))
    }

    private fun createMapper(): ObjectMapper {
        val result = ObjectMapper()
        result.registerModule(KotlinModule())
        return result
    }
}
